tools/workspace-plugin/src/executors/build/lib/babel.ts

```diff
@@ -0,0 +1,125 @@
+import { globSync } from 'fast-glob';
+import { logger } from '@nx/devkit';
+
+import { processAsyncQueue, type NormalizedOptions } from './shared';
+import { compileSwc } from './swc';
+
+const EOL_REGEX = /\r?\n/g;
+
+function addSourceMappingUrl(code: string, sourceMapLocation: string): string {
+  // Babel keeps stripping this comment, even when correct option is set. Adding manually.
+  return code + '\n//# sourceMappingURL=' + sourceMapLocation;
+}
+
+export function hasStylesFilesToProcess(normalizedOptions: NormalizedOptions) {
+  const files = globSync('**/*.styles.ts', { cwd: normalizedOptions.absoluteSourceRoot });
+  return files.length > 0;
+}
+
+export async function compileWithGriffelStylesAOT(options: NormalizedOptions, successCallback: () => Promise<boolean>) {
+  const { esmConfig, restOfConfigs } = options.moduleOutput.reduce<{
+    esmConfig: NormalizedOptions['moduleOutput'][number] | null;
+    restOfConfigs: NormalizedOptions['moduleOutput'];
+  }>(
+    (acc, outputConfig) => {
+      if (outputConfig.module === 'es6') {
+        acc.esmConfig = outputConfig;
+        return acc;
+      }
+
+      acc.restOfConfigs.push(outputConfig);
+      return acc;
+    },
+    { esmConfig: null, restOfConfigs: [] },
+  );
+
+  if (!esmConfig) {
+    logger.warn('es6 module output not specified. Skipping griffel AOT...');
+    const compilationQueue = restOfConfigs.map(outputConfig => {
+      return compileSwc(outputConfig, options);
+    });
+    return processAsyncQueue(compilationQueue, successCallback);
+  }
+
+  await compileSwc(esmConfig, options);
+  await babel(esmConfig, options);
+
+  const compilationQueue = restOfConfigs.map(outputConfig => {
+    const overriddenSourceRoot = join(options.workspaceRoot, options.project.root);
+    // we need to override source root to the output path of transpiled ESM+Griffel AOT, because griffel is unable to handle SWC commonjs output
+    // so instead of transpiling TS(ESM) -> JS(COMMONJS), we transpile JS(ESM + griffel AOT) -> JS(COMMONJS)
+    const overriddenAbsoluteSourceRoot = join(overriddenSourceRoot, esmConfig.outputPath);
+
+    return compileSwc(outputConfig, {
+      ...options,
+      absoluteSourceRoot: overriddenAbsoluteSourceRoot,
+    });
+  });
+
+  return processAsyncQueue(compilationQueue, successCallback);
+}
+
+async function babel(esmModuleOutput: NormalizedOptions['moduleOutput'][number], normalizedOptions: NormalizedOptions) {
+  const filesRoot = join(normalizedOptions.absoluteProjectRoot, esmModuleOutput.outputPath);
+  const files = globSync('**/*.styles.js', { cwd: filesRoot });
+
+  if (files.length === 0) {
+    return;
+  }
+
+  logger.log(`Processing griffel AOT with babel: ${files.length} files`);
+
+  for (const filename of files) {
+    const filePath = join(filesRoot, filename);
+
+    const codeBuffer = await readFile(filePath);
+    const sourceCode = codeBuffer.toString().replace(EOL_REGEX, '\n');
```

Comment:
```suggestion
    const sourceCode = (await readFile(filePath)).toString().replace(EOL_REGEX, '\n');
    const result = await processFile(filePath, sourceCode, BABEL_OPTIONS);
```

Reply:
as mentioned - copy pasted code https://github.com/microsoft/fluentui/pull/32439/files#r1761007131

in general doing too much one line might not be most easies code to read - there are 2 distinctive things happening which makes sense to represent that in 2 varialbes.